src/uffd_handler/src/common.rs

```diff
@@ -30,3 +30,53 @@ pub fn parse_unix_stream(stream: &UnixStream) -> Result<(File, String), StreamEr
 
     Ok((file, body))
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io::Write;
+    use std::os::unix::io::AsRawFd;
+
+    use super::*;
+
+    #[test]
+    fn test_parse_unix_stream() {
+        let (mut sender, receiver) = UnixStream::pair().unwrap();
+        let expected_body = "test body".to_string();
+        let fd = sender.socket_fd();
+
+        // Test case: successfully receive and parse a message body and file descriptor from sender.
```

Comment:
Here and below: would it be possible to have separate tests instead? That could be more readable/modifiable and will provide more granular feedback.